src/Dictionaries/CacheDictionary.cpp

```diff
@@ -251,1032 +262,435 @@ void CacheDictionary::isInConstantVector(const Key child_id, const PaddedPODArra
         out[i] = std::find(ancestors.begin(), ancestors.end(), ancestor_ids[i]) != ancestors.end();
 }
 
-ColumnPtr CacheDictionary::getColumn(
-    const std::string & attribute_name,
-    const DataTypePtr & result_type,
-    const Columns & key_columns,
-    const DataTypes &,
-    const ColumnPtr default_values_column) const
+template <DictionaryKeyType dictionary_key_type>
+void CacheDictionary<dictionary_key_type>::setupHierarchicalAttribute()
 {
-    ColumnPtr result;
-
-    PaddedPODArray<Key> backup_storage;
-    const auto & keys = getColumnVectorData(this, key_columns.front(), backup_storage);
-    auto keys_size = keys.size();
-
-    auto & attribute = getAttribute(attribute_name);
-    const auto & dictionary_attribute = dict_struct.getAttribute(attribute_name, result_type);
-
-    auto type_call = [&](const auto &dictionary_attribute_type)
+    /// TODO: Move this to DictionaryStructure
+    for (const auto & attribute : dict_struct.attributes)
```

Comment:
?

Reply:
Comment is for next refactoring pull requests related to hierarchy handling.